tests/cpptests/micro-benchmarks/benchmark_idlist_iterator.cpp

```diff
@@ -39,64 +41,65 @@ class BM_IdListIterator : public benchmark::Fixture {
       }
     }
     std::sort(docIds.begin(), docIds.end());
+    docIds.erase(std::unique(docIds.begin(), docIds.end()), docIds.end());
+
+    if constexpr (std::is_same_v<IteratorType, QueryIterator>) {
+      iterator_base = IT_V2(NewIdListIterator)(docIds.data(), docIds.size(), 1.0);
+    } else if constexpr (std::is_same_v<IteratorType, IndexIterator>) {
+      iterator_base = NewIdListIterator(docIds.data(), docIds.size(), 1.0);
+    }
+  }
+
+  void TearDown(::benchmark::State &state) {
+    iterator_base->Free(iterator_base);
   }
 };
-bool BM_IdListIterator::initialized = false;
 
-// Translation - exponential range from 2 to 20 (double each time), then 25, 50, 75, and 100.
-// This is the number of docIds to iterate on in each scenario
-#define DOCIDS_SCENARIOS() RangeMultiplier(2)->Range(2, 20)->DenseRange(25, 100, 25)
+template <typename IteratorType>
+bool BM_IdListIterator<IteratorType>::initialized = false;
+
+#define DOCIDS_SCENARIOS() Arg(10)->Arg(100)->Arg(1000)->Arg(100000)
```

Comment:
For the Read benchmark, it shouldn't have any effect, so I suggest a high number so we don't perform the rewind too often (it will hurt the performance)
For skipping, I think we can have a parameter for how much to jump, not sure if it's interesting though.
Remember that the benchmark gives us the average time it takes to perform a single read/skip, so there is no need to make the number of results a parameter if we don't expect it to affect the performance of a single op